### DISCUSSION TITLE
Display station name on subway entrances

### DISCUSSION THREAD

**Discussion by nlehuby** [organicmaps/organicmaps#1216] [Ideas]
It is a good practice not to repeat the name of the station in the name of the subway entrance, and to add it instead into a `stop_area` relation: https://wiki.openstreetmap.org/wiki/Tag:railway%3Dsubway_entrance

So it could be great to display the name of the relation(s) that contains the entrance when you click on it.

**Comment by biodranik** [comment]
Sorry, it's not very clear for me. Can you please show some screenshots of how is it now ("bad"), and describe/draw/show some other apps screenshots how it should be ("good")?

**Comment by nlehuby** [comment]
Here is a subway entrance:
![image](https://user-images.githubusercontent.com/919962/131864936-76267d3d-ed92-40b8-a013-dcd2a4c15702.png)

(https://www.openstreetmap.org/node/1003722008)

As you can see, the name of the entrance is displayed, but not the name of the subway station that can be reached from this entrance.

OsmAnd is somewhat better, it displays the name of the subway station in the panel and the name and ref of the subway_entrance on the map
![image](https://user-images.githubusercontent.com/919962/131865597-18e7ca05-88e6-4623-8b5d-71c31a8df972.png)

**Comment by AdriDevelopsThings** [comment]
Maybe I could add some thoughts I wanted to put into a new discussion but then noticed that this discussion here already exists and could be added here:

Today I was out and about in Hamburg and noticed a misleading information that was shown in Organic Maps. Let me show you a screenshot of the map:
<img width="473" height="597" alt="Screenshot 2026-04-14 at 20 29 57" src="https://github.com/user-attachments/assets/6a2603d0-d121-4711-b56e-35e7f8c798af" />
You see there is an underground station and it's called *Lutterothstraße* but the big name on the entrance shows *Eidelstedter Weg* which is the name of the entrance of the station. The problem in my opinion is that the name is shown in a way that at least could be interpreted as station name instead of entrance name.
Now I want to know from you what is your opinion about this? Was your first thought that the station name is *Eidelstedter Weg* as I thought?

**Reply by biodranik** [reply]
What is in OpenStreetMap.org ? 

Validator shows some unused stations, but they are likely unrelated: https://cdn.organicmaps.app/subway/germany.html

**Reply by AdriDevelopsThings** [reply]
The subway entrance (railway=subway_entrance) has the name *Eidelstedter Weg*. Besides that there is a relation between the entrance and the stop area (public_transport=stop_area). Here is a link to the entrance: https://www.openstreetmap.org/node/249258586.
The data here seems to follow the recommendation that is documented here https://wiki.openstreetmap.org/wiki/Tag:railway%3Dsubway_entrance for the new. (Name of entrance should not be the name of the subway station and instead there should be a relation between the entrance and the stop area).
